### DISCUSSION TITLE
Persistent Notification for Critical Problem like Reauthorization?

### DISCUSSION THREAD

**Discussion by erenoglu** [abraunegg/onedrive#2497] [Ideas]
I'm running onedrive using a systemd user service. Works good. 
Today just out of curiosity I wanted to check the logs and did a "systemd --user status onedrive" command and saw this:

> ERROR: Microsoft OneDrive API returned an error with the following message:
>   Error Message:    HTTP request returned status code 401 (Unauthorized)
>   Error Reason:     Access token is empty.
>   Error Timestamp:  2023-09-22T07:27:04
>   API Request ID:   abcd-efgh-..
> 
> ERROR: Check your configuration as your refresh_token may be empty or invalid. You may need to issue a --reauth and re-authorise this client.
> 
Then I stopped the service, did a manual launch of onedrive with --reauth and got new token, it continued working. But if I didn't look at the systemd logs, I would never noticed this till it hits on a file. 

How can we make sure Onedrive shows a persistent notification or better a dialog box that says it needs re-authentication? Over and over again with certain time periods (ie every 5 10 minutes) even if user cancels it? (user can always stop the daemon to prevent more notifications.  

Or a red circle / exclamation tray icon some other way to notify that something is critically wrong?

**Comment by abraunegg** [comment]
@erenoglu 
> How can we make sure Onedrive shows a persistent notification or better a dialog box that says it needs re-authentication? Over and over again with certain time periods (ie every 5 10 minutes) even if user cancels it? (user can always stop the daemon to prevent more notifications.
>
> Or a red circle / exclamation tray icon some other way to notify that something is critically wrong?

If you are using the GUI (https://github.com/bpozdena/OneDriveGUI) - that would have to be raised there. Potentially raise a feature request for that sort of notification.

The issue itself manifests from your local filesystem - where the reading|writing of the token get's out-of-sync. I saw this heavily in the development of v2.5.0-alpha-0, it would be worth you testing | trying out v2.5.0-alpha-1 for a comparison for this issue.

**Comment by erenoglu** [comment]
I am not using the GUI, the client was working good on the command line so I gave a shot to the systemd. But I have the GUI app, I'll use it.
I can try the alpha-1.

**Comment by helderdaniel** [comment]
I have the same problem with the current and the previous OneDrive version.
I am using version OneDrive v2.5.9-40-g7b3f7a0 with Ubuntu 24.04 LTS.
I don't use the GUI app either.
The filesystem type is ntfs, using driver:
ntfs-3g 2022.10.3 integrated FUSE 28

**Reply by abraunegg** [reply]
@helderdaniel 
> with Ubuntu 24.04 LTS.

This *is* your contributing factor.

You would have installed, as *some* point the Ubuntu package. This *installs* a *default* systemd service that is operating in the background.

It most likely is *this* erroneous systemd service that is responsible for constant notifications about needing re-authorisation.

Essentially - you will have 1 systemd service running that you use, and 1 that is spinning around needing to be authorised.

So how do you fix this - you remove the erroneous systemd service in the first instance.

**Reply by helderdaniel** [reply]
@abraunegg 
Ok, I checked the services and reconfigured them.

After some days of testing, the issue seems to be fixed.

thx

**Reply by abraunegg** [reply]
Glad you have this resolved now

> ` v2.5.9-40-g7b3f7a0`

Please note you are using a self compiled version. 

Due to authentication changes by Microsoft, a new release is pending in the next 12-24 hours. Whilst it will take package maintainers time to release new packages, as you are using Ubuntu, I would *highly* recommend you switch *now* to configuring the OpenSuSE Build Service repository if you have not already done this - so that when the new release is available, you are able to upgrade the client in the supported manner for Ubuntu platforms.

To use the OpenSuSE Build Service repository, please read: https://github.com/abraunegg/onedrive/blob/master/docs/ubuntu-package-install.md

**Reply by helderdaniel** [reply]
Hi,

I stopped the service and uninstalled the compiled version with: make uninstall

Then, after updating the sources, installed the openSuse package with:

sudo apt install --no-install-recommends --no-install-suggests onedrive

When restarting the service I got the error:

Warning: The unit file, source configuration file or drop-ins of onedrive.service changed on disk. Run 'systemctl --user daemon-reload' to reload units.

So I switched back to the compiled version

**Reply by abraunegg** [reply]
> Warning: The unit file, source configuration file or drop-ins of onedrive.service changed on disk. Run 'systemctl --user daemon-reload' to reload units.

all you needed to do was run this command as instructed by your system: `systemctl --user daemon-reload`

When you install a package, it touches the systemd service files, and systemd detects the 'change'

**Reply by helderdaniel** [reply]
oh right!
ok done thx
